vault/data_source_generic_secret.go

```diff
@@ -109,12 +119,28 @@ func genericSecretDataSourceRead(d *schema.ResourceData, meta interface{}) error
 			dataMap[k] = string(vBytes)
 		}
 	}
-	d.Set("data", dataMap)
+	if err := d.Set("data", dataMap); err != nil {
+		return err
+	}
+
+	if err := d.Set("lease_id", secret.LeaseID); err != nil {
+		return err
+	}
+
+	if err := d.Set("lease_duration", secret.LeaseDuration); err != nil {
+		return err
+	}
 
-	d.Set("lease_id", secret.LeaseID)
-	d.Set("lease_duration", secret.LeaseDuration)
-	d.Set("lease_start_time", time.Now().UTC().Format(time.RFC3339))
-	d.Set("lease_renewable", secret.Renewable)
+	if err := d.Set("lease_renewable", secret.Renewable); err != nil {
+		return err
+	}
 
+	if v, ok := d.GetOkExists("with_lease_start_time"); ok {
```

Comment:
I recently discovered that you can use `GetOk()` on a boolean field, if it has a default value. It seems to have worked for me. May want to verify this however.